pkg/apis/serving/v1alpha2/util.go

```diff
@@ -0,0 +1,87 @@
+package v1alpha2
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/apis/core/validation"
+)
+
+func setResourceRequirementDefaults(requirements *v1.ResourceRequirements) {
+	if requirements.Requests == nil {
+		requirements.Requests = v1.ResourceList{}
+	}
+
+	if _, ok := requirements.Requests[v1.ResourceCPU]; !ok {
+		requirements.Requests[v1.ResourceCPU] = DefaultCPU
+	}
+	if _, ok := requirements.Requests[v1.ResourceMemory]; !ok {
+		requirements.Requests[v1.ResourceMemory] = DefaultMemory
+	}
+
+	if requirements.Limits == nil {
+		requirements.Limits = v1.ResourceList{}
+	}
+
+	if _, ok := requirements.Limits[v1.ResourceCPU]; !ok {
+		requirements.Limits[v1.ResourceCPU] = DefaultCPU
+	}
+	if _, ok := requirements.Limits[v1.ResourceMemory]; !ok {
+		requirements.Limits[v1.ResourceMemory] = DefaultMemory
+	}
+}
+
+func transferToCoreResourceRequirements(rr *v1.ResourceRequirements) *core.ResourceRequirements {
```

Comment:
nit: transfer is a bit odd name, maybe just `toCoreResourceRequirements` ?

Reply:
OK, I can update.